### PR TITLE
Address issues with SemanticModel behavior around TupleElementSyntax.

### LINKED ISSUE #14881: Tuple member highlighting and find refs gets confused if there's a local with same name
**Version Used**: 

**Steps to Reproduce**:

```
using System;

static class Program
{
    static void Main()
    {
        (int foo, int bar) tuple;
        int bar;

        tuple = (5, 6);
        tuple.bar = 23;
        bar = 10;

        Console.WriteLine(tuple.bar);
        Console.WriteLine(bar);
    }
}
```

Put you cursor on the various `bar`s. See what gets highlighted, and what's shown if Find References is invoked.

**Expected Behavior**: The highlighter and find references logic should always be able to select the right `bar`s.

**Actual Behavior**: In most cases, the highlighting and find refs logic gets very confused and highlights both `bar` declarations. It only seems to be correct when the cursor is on `tuple.b$$ar` expressions.

Possibly related to #10576.

### LINKED ISSUE #15476: GetDeclaredSymbol should work for tuple elements in a TupleTypeSyntax
This is related to https://github.com/dotnet/roslyn/issues/14881

We should make sure that GetDeclaredSymbol  API works for tuple elements in TupleTypeSyntax.

NOTE: this applies to both VB and C#

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #14881.
Fixes #15476.

@dotnet/roslyn-compiler, @VSadov, @jcouv Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
CC @AnthonyDGreen, @gafter, @jaredpar For public API changes.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -6647,11 +6647,11 @@ private TupleTypeSyntax ParseTupleType()
         private TupleElementSyntax ParseTupleElement()
         {
             var type = ParseType();
-            IdentifierNameSyntax name = null;
+            SyntaxToken name = null;
 
             if (CurrentToken.Kind == SyntaxKind.IdentifierToken)
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: This test should probably be `if (this.IsTrueIdentifier())`

File: src/Compilers/CSharp/Portable/Syntax/Syntax.xml [hunk 2]
```diff
@@ -271,11 +271,11 @@
         <summary>Gets the type of the tuple element.</summary>
       </PropertyComment>
     </Field>
-    <Field Name="Name" Type="IdentifierNameSyntax" Optional="true">
+    <Field Name="Name" Type="SyntaxToken" Optional="true">
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Syntax/Syntax.xml]: This field should probably be renamed `Identifier` to be consistent with convention.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler, @VSadov, @jcouv Please review.

**Pull request merged**